### DISCUSSION TITLE
Getting null (Error 500) when signUp a new user

### DISCUSSION THREAD

**Discussion by soleuil** [supabase/supabase#10195] [Questions] [answered]
I am not sure what is wrong but it is probably something I am not doing properly (beginner here). I have "providers.js" file with the different functions to register, login, logout... I am having issues with a new user registering. I believe it was working just yesterday when I created the piece of code but now I am getting null in the data. The code is this:

```
export const registerUserWithEmailPassword = async ({ email, password }) => {

    try {

        const { data, error } = await supabase.auth.signUp({ email, password })
        const { id } = data.user
        
        console.log( data )

        return {
            ok: true, id, email
        }

    } catch (error) {
        console.log(error);
        return { ok: false, errorMessage: error.message }
    }
}
```

Data is null so I am not getting the user.id to move on. Anything I should change?

**Comment by GaryAustin1** [comment, marked as answer]
Do you have a trigger function on auth.users table for creating a public copy of user data?  This is a very common source of errors that will return at 500.

You also might look in the Postgres and Auth logs in the dashboard for errors.

**Comment by soleuil** [comment]
Thank you so much for your answer. I do have a trigger function for creating a profiles table. I just added some additional fields in the trigger different from the template. Now I tried removing it and creating it all over again but the error persists and I don't know how to fix it. In the Auth logs I have the error 500 for the attempts to signUp but it does not give more details. How could I fix it?

**Reply by soleuil** [reply]
Ok. As I saw that there was some issues with the trigger, I tried with a brand new project and it is working just fine. As I recently created the other one it was not much loss changing over a new one. I hope I don't make the same mistake again. Thanks.
